packages/arb-token-bridge-ui/src/components/TransferPanel/EstimatedGas.tsx

```diff
@@ -0,0 +1,137 @@
+})
+
+function StyledLoader() {
+  return (
+    <span className="flex justify-end">
+      <Loader size="small" />
+    </span>
+  )
+}
+
+export function EstimatedGas({ chain }: { chain: 'parent' | 'child' }) {
+  const {
+    app: { isDepositMode, selectedToken }
+  } = useAppState()
+  const { l1, l2 } = useNetworksAndSigners()
+  const { parentLayer, layer: childLayer } = useChainLayers()
+  const { ethToUSD } = useETHPrice()
+  const nativeCurrency = useNativeCurrency({ provider: l2.provider })
+  const parentChainNativeCurrency = useNativeCurrency({ provider: l1.provider })
+  const isParentChain = chain === 'parent'
+  const {
+    gasSummaryStatus,
+    gasSummary: { estimatedL1GasFees, estimatedL2GasFees }
+  } = useGasSummaryStore()
+  const parentChainName = getNetworkName(l1.network.id)
+  const childChainName = getNetworkName(l2.network.id)
+  const isBridgingEth = selectedToken === null && !nativeCurrency.isCustom
+  const showPrice = useMemo(
+    () => isBridgingEth && !isNetwork(l1.network.id).isTestnet,
+    [isBridgingEth, l1.network.id]
+  )
+  const layer = isParentChain ? parentLayer : childLayer
+
+  const isWithdrawalParentChain = !isDepositMode && isParentChain
+
+  const estimatedGasFee = useMemo(() => {
+    if (!isDepositMode && !isParentChain) {
+      return estimatedL1GasFees + estimatedL2GasFees
+    }
+    return isParentChain ? estimatedL1GasFees : estimatedL2GasFees
+  }, [estimatedL1GasFees, estimatedL2GasFees, isDepositMode, isParentChain])
+
+  const layerGasFeeTooltipContent = (layer: ChainLayer) => {
+    if (!isDepositMode) {
+      return null
+    }
+
+    const { isOrbitChain: isDepositToOrbitChain } = isNetwork(l2.network.id)
+
+    return depositGasFeeTooltip({
+      parentChainName,
+      childChainName,
+      depositToOrbit: isDepositToOrbitChain
+    })[layer]
+  }
+
+  if (isWithdrawalParentChain) {
+    return (
+      <div
+        className={twMerge(
+          'grid items-center rounded-md bg-white/25 px-3 py-2 text-xs font-light text-white'
+        )}
+      >
+        You’ll have to pay {parentChainName} gas fee upon claiming.
+      </div>
+    )
+  }
+
+  return (
+    <div
+      className={twMerge(
+        'grid items-center rounded-md bg-white/25 px-3 py-2 text-right text-xs font-light text-white',
+        showPrice ? 'grid-cols-3' : 'grid-cols-2'
+      )}
+    >
+      <div className="flex flex-row items-center gap-1">
+        <span className="text-left">
+          {isParentChain ? parentChainName : childChainName} gas
+        </span>
+        <Tooltip content={layerGasFeeTooltipContent(layer)}>
+          <InformationCircleIcon className="h-4 w-4" />
+        </Tooltip>
+      </div>
+      {gasSummaryStatus === 'loading' ? (
+        <>
+          {showPrice && <span />}
+          <StyledLoader />
+        </>
+      ) : (
+        <>
+          <span className="text-right tabular-nums">
+            {formatAmount(estimatedGasFee, {
+              symbol: isParentChain
```

Comment:
Parent chain will always have ETH as the native currency, unless there are some cases unknown to us yet such as L4s.

Do we need these checks in place, or can we just check for `nativeCurrency`?

Reply:
Actually if it's for scalability, it doesn't hurt. Just making sure

Reply:
we used to show a breakdown of batch posting fee + execution fee and so we were showing `(?) Xai + (?) ETH` and this PR inherited it
I will clean this up